firestore/distributed_counters.rb

```diff
@@ -0,0 +1,96 @@
+# Copyright 2020 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def create_counter project_id:, num_shards:
+  # project_id = "Your Google Cloud Project ID"
+  # num_shards = "Number of shards for distributed counter"
+
+  require "google/cloud/firestore"
+
+  firestore = Google::Cloud::Firestore.new project_id: project_id
+  # [START fs_create_counter]
+  shards_ref = firestore.col "shards"
+
+  # Initialize each shard with count=0
+  num_shards.times do |i|
+    shards_ref.doc(i).set(count: 0)
+  end
+
+  # [END fs_create_counter]
+  puts "Distributed counter shards collection created."
```

Comment:
```suggestion
  # [START fs_create_counter]
  # project_id = "Your Google Cloud Project ID"
  # num_shards = "Number of shards for distributed counter"

  require "google/cloud/firestore"

  firestore = Google::Cloud::Firestore.new project_id: project_id

  shards_ref = firestore.col "shards"

  # Initialize each shard with count=0
  num_shards.times do |i|
    shards_ref.doc(i).set(count: 0)
  end

  puts "Distributed counter shards collection created."
  # [END fs_create_counter]
```

generally, snippets are meant to be copy-paste runnable, so both the requires and the client initialization should be included in the region tags

Reply:
@TheRoyalTnetennba  updated code.